text/XXXX-replace_publish_clone_with_push_pull.md

```diff
@@ -0,0 +1,160 @@
+- Feature Name: Dataset Push & Pull
+- Start Date: <!-- (fill me in with today's date, YYYY-MM-DD) -->
+- RFC PR: <!-- (leave this empty) -->
+- Issue: <!-- (leave this empty) -->
+
+# Summary
+[summary]: #summary
+
+Replace `publish/add` with `push/pull`, which default to pushing all data, pulling only the latest version. A new `clone` command combines `pull` with `checkout`.
+
+# Motivation
+[motivation]: #motivation
+
+This RFC addresses two problems with the way we move versions around in qri:
+1. datasets published to the registry are creating weird states where users can't get versions
```

Comment:
I might just be getting lost because the solutions for this are mixed in with renaming things here, but after reading the rest of this doc, I don’t feel totally clear on exactly what the major issues with publishing/pushing are (or what kinds of weird states are getting created) or exactly how this proposal addresses them. I know I’ve run into at least one of these, but I don’t know what broader scope of issues looks like. (The changes described here still definitely seem like good improvements and clarifications, though!)

Reply:
I also think we are on the right track.  I've found it hard to understand what exactly was published when I publish, or what I got when I used "add".  I'd also suggest the CLI and Desktop reinforce what is happening.

```
> qri push me/some_dataset
pushing 28 versions of chriswhong/some_dataset...
X of XX blocks pushed for version 12 August 2019 (Qmss4h552h3j2h33)...

> qri pull me/some_dataset
pulling latest version of chriswhong/some_dataset | 13 August 2019 (Qmfie4kjn2di3223)...
There are 17 prior versions. Use --all get all, or use the full dataset reference to get a specific version
```

Reply:
Super helpful early feedback. I wrote the solutions before the motivation section (problem statement). I'll add some concrete examples